mono/metadata/reflection.c

```diff
@@ -7156,6 +7156,7 @@ mono_module_file_get_object_checked (MonoDomain *domain, MonoImage *image, int t
 	MONO_OBJECT_SETREF (res, fqname, mono_string_new (domain, name));
 	MONO_OBJECT_SETREF (res, name, mono_string_new (domain, name));
 	MONO_OBJECT_SETREF (res, scopename, mono_string_new (domain, name));
+	// Shouldn't this be & instead of &&?
 	res->is_resource = cols [MONO_FILE_FLAGS] && FILE_CONTAINS_NO_METADATA;
 	res->token = mono_metadata_make_token (MONO_TABLE_FILE, table_index + 1);
```

Comment:
Agreed, this is extracting a column from the database that should have two possible values FILE_CONTAINS_METADATA or FILE_CONTAINS_NO_METADATA.

While currently harmless, as the only two possible values are 0 and 1, this is documented as being a flags field, and if this got more data in the future, the above would break.